src/JWT/Builder/JwtBuilder.cs

```diff
@@ -247,6 +249,22 @@ public string Encode()
             return _encoder.Encode(_jwt.Header, _jwt.Payload, _secrets?[0]);
         }
 
+        public string Encode<T>(T payload)
+        {
+            EnsureCanEncode();
+
+            var dic = payload.GetType()
+                             .GetProperties(BindingFlags.Instance | BindingFlags.Public)

```

Comment:
I think it should be rather `typeof(T)` which would be a simple syntax sugar on top of `Encode(Type payloadType, object payload)` as it's already in `IJwtDecoderExtensions`, check it out.